Legacy/bonej/src/main/java/org/bonej/plugins/ParticleCounter.java

```diff
@@ -552,13 +551,17 @@ else if (workArray[z][arrayIndex] == FORE &&
 	 * @param lut a look-up table.
 	 * @return total number of duplicate values found.
 	 */
-	private static long countDuplicates(final int[] counter,
-		final List<HashSet<Integer>> map, final int[] lut)
+	private static int countDuplicates(int[] counter,
+		final ArrayList<HashSet<Integer>> map, final int[] lut)
 	{
+		//reset to 0 the counter array
+		final int l = counter.length;
+		counter = new int[l];
```

Comment:
Is this clearing of the counter array what prevents the AIOOB from happening in the "special cases" you mention?

Reply:
Maybe. There was also somewhere else an instantiation that got moved inside a while(){}, which might have been responsible.